From 7206bb2a9b655be3bdb66c6cf03aa504817ed240 Mon Sep 17 00:00:00 2001
From: Lars-Peter Clausen <lars@metafoo.de>
Date: Tue, 14 Aug 2018 09:48:59 +0200
Subject: [PATCH] Fix building outside git repository

Currently when building the project on a machine that has git installed,
but the source is not inside a git repository cmake fails with the
following error.

	CMake Error at CMakeLists.txt:35 (if):
	  if given arguments:

		"STREQUAL" "/home/lars/libad9361-iio"

This is due to LIBAD9361_GIT_REPO variable being empty. In order to handle
empty variables in CMake in a string comparison they need to be
encapsulated in quotes.

Signed-off-by: Lars-Peter Clausen <lars@metafoo.de>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7af5444..8d04ba0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@ if (GIT_FOUND)
 		OUTPUT_STRIP_TRAILING_WHITESPACE
 	)
 
-if (${LIBAD9361_GIT_REPO} STREQUAL ${CMAKE_CURRENT_SOURCE_DIR})
+if ("${LIBAD9361_GIT_REPO}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
 		execute_process(
 			COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
 			WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
